oracle/src/events/processAMBAffirmationRequests/estimateGas.js

```diff
@@ -28,6 +28,12 @@ async function estimateGas({ web3, homeBridge, validatorContract, message, addre
     const messageHash = web3.utils.soliditySha3(message)
     const senderHash = web3.utils.soliditySha3(address, messageHash)
 
+    // Check if msg is approved by Hashi
+    const isApprovedByHashi = await homeBridge.methods.isApprovedByHashi(messageHash)
```

Comment:
i think you have to check if hashi is mandatory here

Reply:
Solve in [b688329](https://github.com/gnosischain/tokenbridge/pull/5/commits/b688329b9028bdd7d43ee47bddd045ebb0ae2d27)